Results_Engine/Query/MapResults.cs

```diff
@@ -37,14 +38,17 @@ public static partial class Query
         /****           Public Methods                  ****/
         /***************************************************/
 
+        [PreviousVersion("5.1", "BH.Engine.Results.Query.MapResults(System.Collections.Generic.IEnumerable<BH.oM.Base.IBHoMObject>, System.Collections.Generic.IEnumerable<BH.oM.Analytical.Results.IResult>, System.String, System.Type, System.Collections.Generic.List<System.String>)")]
+        [PreviousInputNames("objectIdentifier", "identifier")]
+        [PreviousInputNames("resultIdentifier", "whichId")]
         [Description("Matches results to BHoMObjects. The output consists of a list of items corresponding to the list of BHoMObjects supplied. Each item in the output list is a list of results matching the relevant BHoMObject. The results will be matched by the ID of the object stored in the Fragments and the ObjectId of the result. If no results are found, an empty list will be provided. Note that NO compatibility check between object type and result type will be made.")]
         [Input("objects", "The objects to find results for.")]
         [Input("results", "The collection of results to search in.")]
-        [Input("whichId", "The name of the object identifier to group results by. Defaults to ObjectId.")]
-        [Input("identifier", "The type of IAdapterId fragment to be used to extract the object identification, i.e. which fragment type to look for to find the identifier of the object. If no identifier is provided, the object will be scanned an IAdapterId to be used.")]
+        [Input("resultIdentifier", "Property of the IResult used to map the result to the Object. Defaults to ObjectId.")]
+        [Input("objectIdentifier", "Should either be a string specifying what property on the object that should be used to map the objects to the results, or a type of IAdapterId fragment to be used to extract the object identification, i.e. which fragment type to look for to find the identifier of the object. If no identifier is provided, the object will be scanned an IAdapterId to be used.")]
         [Input("caseFilter", "Optional filter for the case. If nothing is provided, all cases will be used.")]
         [Output("results", "Results as a List of List where each inner list corresponds to one BHoMObject based on the input order.")]
-        public static List<List<TResult>> MapResults<TResult, TObject>(this IEnumerable<TObject> objects, IEnumerable<TResult> results, string whichId = nameof(IObjectIdResult.ObjectId), Type identifier = null, List<string> caseFilter = null) 
+        public static List<List<TResult>> MapResults<TResult, TObject>(this IEnumerable<TObject> objects, IEnumerable<TResult> results, string resultIdentifier = nameof(IObjectIdResult.ObjectId), object objectIdentifier = null, List<string> caseFilter = null)
```

Comment:
I'm considering the accuracy of this method name and inputs ordering. Because the "main subject" (the input that is specified with the `this` keyword) seems to be `objects`, just by considering that I would say that `MapObjects()` could be a more accurate name, especially because we are already inside the namespace `Results`.

However, `MapObjects()` does not seem to be what the method actually does. So if we keep the name `MapResults()` maybe we should consider changing the order of the inputs to be `(this IEnumerable<TResult> results, IEnumerable<TObject> objects, ...` instead?

Personally I would prefer an even more explicit solution, with swapped inputs:
`MapResultsToObjects(this IEnumerable<TResult> results, IEnumerable<TObject> objects, ...)`.